### PR TITLE
✨ replace the feature flag manager-based module availability checker with a configuration-based module availability checker

### PR DESCRIPTION
**Pull request by kamilbaczek**
## 📋 Description
Replace the feature flag manager-based module availability checker with a configuration-based module availability checker. ⚙️

This change shifts the mechanism for checking module availability from using feature flags to using configuration settings, simplifying the process and improving maintainability. 📊

## 📦 PR Includes

- [ ] Feature added 🆕
- [ ] Bug fix 🐛
- [X] Code refactor 🔄
- [ ] Documentation update 📚
- [ ] Tests added/updated 🧪
- [ ] Other: (describe) 

## 💡 Additional Notes

No additional notes

### CODE REVIEW

File: Chapter-2-modules-separation/Src/Common/Fitnet.Common.Core/Fitnet.Common.Core.csproj [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.1" />
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Fitnet.Common.Core/Fitnet.Common.Core.csproj:4]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Common/Fitnet.Common.Api/Fitnet.Common.Api.csproj [hunk 2]
```diff
@@ -16,8 +16,9 @@
     </ItemGroup>
 
     <ItemGroup>
-        <PackageReference Include="FluentValidation" Version="11.7.1" />
-        <PackageReference Include="FluentValidation.DependencyInjectionExtensions" Version="11.7.1" />
+        <PackageReference Include="FluentValidation" Version="11.9.1" />
+        <PackageReference Include="FluentValidation.DependencyInjectionExtensions" Version="11.9.1" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Fitnet.Common.Api/Fitnet.Common.Api.csproj:21]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Common/Fitnet.Common.Infrastructure/Modules/ModuleAvailabilityChecker.cs [hunk 3]
```diff
@@ -1,48 +1,15 @@
...
-using Microsoft.FeatureManagement;
 using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.DependencyInjection;
 
-public sealed class ModuleAvailabilityChecker : IDisposable
+public static class ModuleAvailabilityChecker
 {
-    private readonly IFeatureManager _featureManager;
-    private readonly ServiceProvider _serviceProvider;
+    private const string EnabilityConfigName = "Enabled";
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Fitnet.Common.Infrastructure/Modules/ModuleAvailabilityChecker.cs:7]: Rename Enability to Availability

File: Chapter-2-modules-separation/Src/Common/Tests/Fitnet.Common.Infrastructure.IntegrationTests/Fitnet.Common.Infrastructure.IntegrationTests.csproj [hunk 4]
```diff
@@ -6,6 +6,7 @@
 
     <ItemGroup>
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Tests/Fitnet.Common.Infrastructure.IntegrationTests/Fitnet.Common.Infrastructure.IntegrationTests.csproj:9]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Common/Tests/Fitnet.Common.Core.UnitTests/Fitnet.Common.Core.UnitTests.csproj [hunk 5]
```diff
@@ -9,9 +9,10 @@
     </ItemGroup>
 
     <ItemGroup>
-      <PackageReference Include="FluentAssertions" Version="6.10.0" />
-      <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
-      <PackageReference Include="xunit" Version="2.4.2" />
+      <PackageReference Include="FluentAssertions" Version="6.12.0" />
+      <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+      <PackageReference Include="xunit" Version="2.8.0" />
+      <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Tests/Fitnet.Common.Core.UnitTests/Fitnet.Common.Core.UnitTests.csproj:15]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Common/Tests/Fitnet.Common.Api.UnitTests/Fitnet.Common.Api.UnitTests.csproj [hunk 6]
```diff
@@ -9,10 +9,11 @@
...
     <ItemGroup>
-        <PackageReference Include="FluentAssertions" Version="6.10.0" />
-        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
-        <PackageReference Include="NSubstitute" Version="5.0.0" />
-        <PackageReference Include="xunit" Version="2.4.2" />
+        <PackageReference Include="FluentAssertions" Version="6.12.0" />
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+        <PackageReference Include="NSubstitute" Version="5.1.0" />
+        <PackageReference Include="xunit" Version="2.8.0" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Tests/Fitnet.Common.Api.UnitTests/Fitnet.Common.Api.UnitTests.csproj:16]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Common/Fitnet.Common.IntegrationTests/Fitnet.Common.IntegrationTests.csproj [hunk 7]
```diff
@@ -6,25 +6,26 @@
...
+        <PackageReference Include="xunit" Version="2.8.0" />
+        <PackageReference Include="xunit.analyzers" Version="1.13.0" />
+        <PackageReference Include="xunit.categories" Version="2.0.8" />
+        <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="8.0.5" />
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+        <PackageReference Include="coverlet.collector" Version="6.0.2">
             <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
             <PrivateAssets>all</PrivateAssets>
         </PackageReference>
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Fitnet.Common.IntegrationTests/Fitnet.Common.IntegrationTests.csproj:28]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Common/Fitnet.Common.Infrastructure/Fitnet.Common.Infrastructure.csproj [hunk 8]
```diff
@@ -1,11 +1,14 @@
...
         <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
-        <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />
-        <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="7.0.0" />
-        <PackageReference Include="Microsoft.FeatureManagement" Version="2.5.1" />
+        <PackageReference Include="Microsoft.Extensions.Configuration" Version="8.0.0" />
+        <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="8.0.0" />
+        <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="8.0.1" />
+        <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="8.0.0" />
+        <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.1" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Common/Fitnet.Common.Infrastructure/Fitnet.Common.Infrastructure.csproj:11]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.Application/Fitnet.Contracts.Application.csproj [hunk 9]
```diff
@@ -1,7 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <ItemGroup>
-    <PackageReference Include="JetBrains.Annotations" Version="2022.3.1" />
-    <PackageReference Include="MediatR" Version="12.0.1" />
+    <PackageReference Include="JetBrains.Annotations" Version="2023.3.0" />
+    <PackageReference Include="MediatR" Version="12.2.0" />
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.Application/Fitnet.Contracts.Application.csproj:5]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj [hunk 10]
```diff
@@ -6,5 +6,9 @@
   <ItemGroup>
     <InternalsVisibleTo Include="EvolutionaryArchitecture.Fitnet.Contracts.Core.UnitTests" />
   </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.Core/Fitnet.Contracts.Core.csproj:11]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.Infrastructure/Fitnet.Contracts.Infrastructure.csproj [hunk 11]
```diff
@@ -5,10 +5,12 @@
...
   <ItemGroup>
-    <PackageReference Include="MediatR" Version="12.0.1" />
+    <PackageReference Include="MediatR" Version="12.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.0" />
-    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="7.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.5" />
+    <PackageReference Include="Npgsql" Version="8.0.3" />
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="8.0.4" />
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.Infrastructure/Fitnet.Contracts.Infrastructure.csproj:13]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.IntegrationEvents/Fitnet.Contracts.IntegrationEvents.csproj [hunk 12]
```diff
@@ -4,4 +4,8 @@
       <ProjectReference Include="..\..\Common\Fitnet.Common.Infrastructure\Fitnet.Common.Infrastructure.csproj" />
     </ItemGroup>
 
+    <ItemGroup>
+      <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Contracts/Fitnet.Contracts.IntegrationEvents/Fitnet.Contracts.IntegrationEvents.csproj:8]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Contracts/Tests/Fitnet.Contracts.Core.UnitTests/Fitnet.Contracts.Core.UnitTests.csproj [hunk 13]
```diff
@@ -5,20 +5,21 @@
...
             <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
         </PackageReference>
-        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
-        <PackageReference Include="coverlet.collector" Version="3.2.0">
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+        <PackageReference Include="coverlet.collector" Version="6.0.2">
             <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
             <PrivateAssets>all</PrivateAssets>
         </PackageReference>
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Contracts/Tests/Fitnet.Contracts.Core.UnitTests/Fitnet.Contracts.Core.UnitTests.csproj:22]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Contracts/Tests/Fitnet.Contracts.IntegrationTests/Fitnet.Contracts.IntegrationTests.csproj [hunk 14]
```diff
@@ -5,9 +5,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Bogus" Version="34.0.2" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
-    <PackageReference Include="NSubstitute" Version="5.0.0" />
+    <PackageReference Include="Bogus" Version="35.5.1" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+    <PackageReference Include="NSubstitute" Version="5.1.0" />
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Contracts/Tests/Fitnet.Contracts.IntegrationTests/Fitnet.Contracts.IntegrationTests.csproj:11]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Offers/Fitnet.Offers.Api/Fitnet.Offers.Api.csproj [hunk 15]
```diff
@@ -15,4 +15,8 @@
         <InternalsVisibleTo Include="EvolutionaryArchitecture.Fitnet.Offers.IntegrationTests" />
     </ItemGroup>
 
+    <ItemGroup>
+      <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Offers/Fitnet.Offers.Api/Fitnet.Offers.Api.csproj:19]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Offers/Fitnet.Offers.DataAccess/Fitnet.Offers.DataAccess.csproj [hunk 16]
```diff
@@ -1,7 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
     <ItemGroup>
         <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
-        <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.0" />
-        <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="7.0.0" />
+        <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.5" />
+        <PackageReference Include="Npgsql" Version="8.0.3" />
+        <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="8.0.4" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Offers/Fitnet.Offers.DataAccess/Fitnet.Offers.DataAccess.csproj:7]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Offers/Tests/Fitnet.Offers.IntegrationTests/Fitnet.Offers.IntegrationTests.csproj [hunk 17]
```diff
@@ -5,9 +5,10 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Bogus" Version="34.0.2" />
-        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
-        <PackageReference Include="NSubstitute" Version="5.0.0" />
+        <PackageReference Include="Bogus" Version="35.5.1" />
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+        <PackageReference Include="NSubstitute" Version="5.1.0" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Offers/Tests/Fitnet.Offers.IntegrationTests/Fitnet.Offers.IntegrationTests.csproj:11]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Passes/Fitnet.Passes.Api/Fitnet.Passes.Api.csproj [hunk 18]
```diff
@@ -18,6 +18,7 @@
   </ItemGroup>
   
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="7.0.10" />
+    <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.5" />
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Passes/Fitnet.Passes.Api/Fitnet.Passes.Api.csproj:22]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Passes/Fitnet.Passes.DataAccess/Fitnet.Passes.DataAccess.csproj [hunk 19]
```diff
@@ -1,7 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.0" />
-    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="7.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="8.0.5" />
+    <PackageReference Include="Npgsql" Version="8.0.3" />
+    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="8.0.4" />
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Passes/Fitnet.Passes.DataAccess/Fitnet.Passes.DataAccess.csproj:7]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Passes/Fitnet.Passes.IntegrationEvents/Fitnet.Passes.IntegrationEvents.csproj [hunk 20]
```diff
@@ -2,4 +2,7 @@
   <ItemGroup>
     <ProjectReference Include="..\..\Common\Fitnet.Common.Infrastructure\Fitnet.Common.Infrastructure.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
+  </ItemGroup>
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Passes/Fitnet.Passes.IntegrationEvents/Fitnet.Passes.IntegrationEvents.csproj:7]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Passes/Tests/Fitnet.Passes.IntegrationTests/Fitnet.Passes.IntegrationTests.csproj [hunk 21]
```diff
@@ -5,9 +5,10 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Bogus" Version="34.0.2" />
-        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
-        <PackageReference Include="NSubstitute" Version="5.0.0" />
+        <PackageReference Include="Bogus" Version="35.5.1" />
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+        <PackageReference Include="NSubstitute" Version="5.1.0" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Passes/Tests/Fitnet.Passes.IntegrationTests/Fitnet.Passes.IntegrationTests.csproj:11]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Reports/Fitnet.Reports/Fitnet.Reports.csproj [hunk 22]
```diff
@@ -7,9 +7,10 @@
       <ProjectReference Include="..\..\Common\Fitnet.Common.Infrastructure\Fitnet.Common.Infrastructure.csproj" />
     </ItemGroup>
     <ItemGroup>
-      <PackageReference Include="Dapper" Version="2.0.123" />
-      <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="7.0.10" />
-      <PackageReference Include="Npgsql" Version="7.0.4" />
+      <PackageReference Include="Dapper" Version="2.1.35" />
+      <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="8.0.5" />
+      <PackageReference Include="Npgsql" Version="8.0.3" />
+      <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Reports/Fitnet.Reports/Fitnet.Reports.csproj:13]: It is defined on the level of Directory.Build.props, please remove it from here

File: Chapter-2-modules-separation/Src/Reports/Tests/Fitnet.Reports.IntegrationTests/Fitnet.Reports.IntegrationTests.csproj [hunk 23]
```diff
@@ -5,8 +5,9 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Bogus" Version="34.0.2" />
-        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
+        <PackageReference Include="Bogus" Version="35.5.1" />
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+        <PackageReference Update="SonarAnalyzer.CSharp" Version="9.25.0.90414" />
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Src/Reports/Tests/Fitnet.Reports.IntegrationTests/Fitnet.Reports.IntegrationTests.csproj:10]: It is defined on the level of Directory.Build.props, please remove it from here

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]
Update the package of SonarAnalyzer on the level of Directory build props.

Btw. I will add an issue to replace configuration with IOptions (we can do it in the next months, not now).

File: Chapter-2-modules-separation/Src/Common/Fitnet.Common.Api/Fitnet.Common.Api.csproj (hunk 2 above)
Comment (kamilbaczek) [reply Chapter-2-modules-separation/Src/Common/Fitnet.Common.Api/Fitnet.Common.Api.csproj]: Done

**Review by meaboutsoftware — APPROVED** [review]

**Pull request merged**